### DISCUSSION TITLE
Use Swift Package Manager to add AdaptiveCards framework in iOS

### DISCUSSION THREAD

**Discussion by marting07** [microsoft/AdaptiveCards#6910] [Q&A]
Hello all,

Right now we're using AdaptiveCards framework for a chat app and we're using cocoapods as a dependency manager. We want to move away from it and use Swift Package Manager (SPM) instead. We spotted that this amazing team added the capability to support this. 

We tried to add the AdaptiveCards framework using SPM via versions (2020.0.0), also pointing to master branch of the repo and we weren’t able to do it, SPM throws some errors.

We started to review open and close issues and discussions (https://github.com/microsoft/AdaptiveCards/issues?q=is%3Aissue+spm), we spotted a comment from @jwoo-msft about using a commit hash. We tried with https://github.com/microsoft/AdaptiveCards/commit/58d7c4152bc39a48087062a543bc180a1b1b1bb9 and https://github.com/microsoft/AdaptiveCards/commit/830fa1fd73efd0c57b958144f58ad7ee74c77139 this last commit seems that worked but we couldn’t find any final guide or we don't know if we're using the correct commit hash.

What would be the correct commit hash we need to include AdaptiveCards framework to our app?
(note sure if we can ask you directly @jwoo-msft?)

**Comment by jwoo-msft** [comment]
@marting07 
It should be `96076818f`
I updated our last release note accordingly. https://github.com/microsoft/AdaptiveCards/releases/tag/21.09
Please let us know if it doesn't work.

**Reply by marting07** [reply]
Thanks a lot for the quick answer @jwoo-msft. We were able to include the framework successfully, unfortunately when we tried to build/run the sample app we got a build error, you can see the screenshot below:

<img width="1144" alt="Screen Shot 2022-02-02 at 3 36 06" src="https://user-images.githubusercontent.com/20322568/152129245-862ed248-9e9e-4a6d-8a97-e5a45f9fba01.png">

<img width="269" alt="Screen Shot 2022-02-02 at 3 36 25" src="https://user-images.githubusercontent.com/20322568/152129269-bfb8244a-12e3-4e43-b197-e0b60067c314.png">

We use the exact same sample app to test the other commit hashes I mentioned in the first question of this thread.

I'll be attentive in case I can help with testing.

**Reply by jwoo-msft** [reply]
@marting07 
We started using framework header after we switched cocoapod's distribution format from framework to source, and looks like that's causing the issue.
I found [this link](https://developer.apple.com/forums/thread/69606) that is similar to this issue.
TLDR directly from the link
```
SOLVED:
I had to manually add the .h files from the framework into the build phase (eg. KSFrameworkRoot.h) and it works nicely.
```

**Reply by marting07** [reply]
@jwoo-msft Thanks for the quick response again and the hints of what I need to solve the issue. I tried the solution mentioned here but I couldn’t solve the issue adding the .h files in the build phase. But I think I found the possible issue.

Taking a look at other .h files in the source code I spotted that we don't have the `<AdaptiveCards/` part of the path for them:

<img width="1144" alt="Screen Shot 2022-02-03 at 4 59 16" src="https://user-images.githubusercontent.com/20322568/152330905-be1901bc-39d3-4577-a39c-2729a6de39c1.png">

But for `ACRImageProperties.h` file we add `<AdaptiveCards/` to try to include `ACOBaseCardElement`:

<img width="1219" alt="Screen Shot 2022-02-03 at 4 59 48" src="https://user-images.githubusercontent.com/20322568/152331391-717e7fb1-8fd2-4a6d-bbee-e1375e1d3cb4.png">

I'm attentive to any comment you have, thanks so much.

**Reply by jwoo-msft** [reply]
we will need to have macro define that put framework headers separate from SPM users. 
For example,
```
#ifdef SWIFT_PACKAGE
#import "ACOBaseCardElement.h"
#else
#import <AdaptiveCards/ACOBaseCardElement.h>
#endif
```

**Reply by marting07** [reply]
I see, thanks a lot for the explanation @jwoo-msft. Then I guess we'll be attentive for that update. Not sure also if we can help in any way?

**Comment by jwoo-msft** [comment]
@marting07 sorry for late response; SPM is not supported.

Thank you!

**Comment by vicajilau** [comment]
After more than 1 year I was wondering if it is planned to support Swift Package Manager at some point @jwoo-msft. Thank you very much!
